engine/class_modules/monk/sc_monk.cpp

```diff
@@ -3811,10 +3868,26 @@ struct breath_of_fire_t : public monk_spell_t
 // Fortifying Brew
 // ==========================================================================
 
+// Niuzao's Protection ==============================================
 struct fortifying_brew_t : brew_t<monk_spell_t>
 {
+  struct niuzaos_protection_t : public monk_absorb_t
+  {
+    niuzaos_protection_t( monk_t *p )
+      : monk_absorb_t( p, "niuzaos_protection", p->talent.conduit_of_the_celestials.niuzaos_protection )
+    {
+      background  = true;
+      target      = p;
+      base_dd_min = p->resources.max[ RESOURCE_HEALTH ] * data().effectN( 2 ).percent();
+      base_dd_max = base_dd_min;
+    }
+  };
+
+  niuzaos_protection_t *absorb;
+
   fortifying_brew_t( monk_t *p, util::string_view options_str )
-    : brew_t<monk_spell_t>( p, "fortifying_brew", p->talent.monk.fortifying_brew.find_override_spell() )
+    : brew_t<monk_spell_t>( p, "fortifying_brew", p->talent.monk.fortifying_brew.find_override_spell() ),
+      absorb( new niuzaos_protection_t( p ) )
```

Comment:
Only instantiate if the talent is known.
Only execute if `absorb` is non-`nullptr`.